src/Config/SessionConfig.php

```diff
@@ -447,40 +447,39 @@ private function locateRegisteredSaveHandlers()
      */
     private function performSaveHandlerUpdate($phpSaveHandler)
     {
-        $knownHandlers = $this->locateRegisteredSaveHandlers();
-
-        if (in_array($phpSaveHandler, $knownHandlers, true)) {
-            $phpSaveHandler = strtolower($phpSaveHandler);
-            set_error_handler([$this, 'handleError']);
-            session_module_name($phpSaveHandler);
-            restore_error_handler();
-            if ($this->phpErrorCode >= E_WARNING) {
+        if (is_string($phpSaveHandler)) {
+            $knownHandlers = $this->locateRegisteredSaveHandlers();
+
+            if (in_array($phpSaveHandler, $knownHandlers, true)) {
+                $phpSaveHandler = strtolower($phpSaveHandler);
+                set_error_handler([$this, 'handleError']);
+                session_module_name($phpSaveHandler);
+                restore_error_handler();
+                if ($this->phpErrorCode >= E_WARNING) {
+                    throw new Exception\InvalidArgumentException(sprintf(
+                        'Error setting session save handler module "%s": %s',
+                        $phpSaveHandler,
+                        $this->phpErrorMessage
+                    ));
+                }
+
+                return $phpSaveHandler;
+            }
+
+            if ((! class_exists($phpSaveHandler)
+                    || ! (in_array(SessionHandlerInterface::class, class_implements($phpSaveHandler)))
```

Comment:
I see this condition was like that before, but I think it can be simplify by using `is_a` method.

Reply:
I tried, but unit tests are not working after change. Need to investigate more. I will do it in another MR.

Reply:
Have you tried `is_a($phpSaveHandler, SessionHandlerInterface::class, true)` ?
I've seen you've tried `instanceof` but this is not gonna work.

Reply:
I tried with is_a but that will break unit tests. Maybe we can fix it in another PR/issue.

Reply:
I've checked and it works for me:
```
            if (! class_exists($phpSaveHandler)
                || ! is_a($phpSaveHandler, SessionHandlerInterface::class, true)
            ) {
```

Reply:
You are right, Thanks. Its changed